datafusion/sql/examples/sql_parser.rs

```diff
@@ -15,6 +15,10 @@
 // specific language governing permissions and limitations
 // under the License.
 
+//! This example shows how to use (only) the DataFusion SQL parser
```

Comment:
I renamed this example (not it is in datafusion-sql not datafusion core) and added some comments about what it does